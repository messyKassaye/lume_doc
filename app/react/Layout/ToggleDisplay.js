import PropTypes from 'prop-types';
import React from 'react';

class ToggleDisplay extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      show: props.open,
    };
    this.show = this.show.bind(this);
    this.hide = this.hide.bind(this);
  }

  show() {
    this.setState({ show: true });
  }

  hide() {
    this.setState({ show: false });
    this.props.onHide();
  }

  render() {
    return (
      <>
        {!this.state.show && <button onClick={this.show}>{this.props.showLabel}</button>}
        {this.state.show && <button onClick={this.hide}>{this.props.hideLabel}</button>}
        {this.state.show && this.props.children}
      </>
    );
  }
}

ToggleDisplay.defaultProps = {
  onHide: () => {},
  showLabel: 'show',
  hideLabel: 'hide',
  open: false,
};

const labelType = PropTypes.oneOfType([PropTypes.string, PropTypes.element]);

ToggleDisplay.propTypes = {
  children: PropTypes.element.isRequired,
  onHide: PropTypes.func,
  showLabel: labelType,
  hideLabel: labelType,
  open: PropTypes.bool,
};

export default ToggleDisplay;
