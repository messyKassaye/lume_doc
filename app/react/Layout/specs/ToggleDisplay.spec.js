import React from 'react';

import { shallow } from 'enzyme';

import ToggleDisplay from '../ToggleDisplay';

describe('ToggleDisplay', () => {
  it('should not render children by default', () => {
    const component = shallow(
      <ToggleDisplay>
        <span>children</span>
      </ToggleDisplay>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render labels passed for buttons', () => {
    const component = shallow(
      <ToggleDisplay showLabel="showLabel" hideLabel="hideLabel">
        <span>children</span>
      </ToggleDisplay>
    );
    expect(component.find('button')).toMatchSnapshot();
    component.setState({ show: true });
    expect(component.find('button')).toMatchSnapshot();
  });

  describe('when passed open true', () => {
    it('should show the children by default', () => {
      const component = shallow(
        <ToggleDisplay open>
          <span>children</span>
        </ToggleDisplay>
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe('on show button click', () => {
    it('should render children passed and hide button', () => {
      const component = shallow(
        <ToggleDisplay>
          <span>children</span>
        </ToggleDisplay>
      );
      component.find('button').simulate('click');
      expect(component).toMatchSnapshot();
    });
  });

  describe('on hide button click', () => {
    it('should render hide children', () => {
      const component = shallow(
        <ToggleDisplay>
          <span>children</span>
        </ToggleDisplay>
      );
      component.find('button').simulate('click');
      component.find('button').simulate('click');
      expect(component).toMatchSnapshot();
    });

    it('should call onHide callback prop', () => {
      const onHide = jasmine.createSpy('onHide');
      const component = shallow(
        <ToggleDisplay onHide={onHide}>
          <span>children</span>
        </ToggleDisplay>
      );
      component.find('button').simulate('click');
      component.find('button').simulate('click');
      expect(onHide).toHaveBeenCalled();
    });
  });
});
