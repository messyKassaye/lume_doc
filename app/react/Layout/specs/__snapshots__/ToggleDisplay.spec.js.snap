// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleDisplay on hide button click should render hide children 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    show
  </button>
</Fragment>
`;

exports[`ToggleDisplay on show button click should render children passed and hide button 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    hide
  </button>
  <span>
    children
  </span>
</Fragment>
`;

exports[`ToggleDisplay should not render children by default 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    show
  </button>
</Fragment>
`;

exports[`ToggleDisplay should render labels passed for buttons 1`] = `
<button
  onClick={[Function]}
>
  showLabel
</button>
`;

exports[`ToggleDisplay should render labels passed for buttons 2`] = `
<button
  onClick={[Function]}
>
  hideLabel
</button>
`;

exports[`ToggleDisplay when passed open true should show the children by default 1`] = `
<Fragment>
  <button
    onClick={[Function]}
  >
    hide
  </button>
  <span>
    children
  </span>
</Fragment>
`;
